src/components/CharacterList.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { useSelector, useDispatch } from 'react-redux';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Link } from 'react-router-dom';
+import { fetchCharactersAction } from '../redux/reducer/reducer';
+
+function CharacterList() {
+  const [charNameInput, setCharNameInput] = useState('');
+  const dispatch = useDispatch();
+
+  const inputCharNameChangeHandler = (e) => {
+    setCharNameInput(e.target.value);
+  };
+
+  useEffect(() => {
+    dispatch(fetchCharactersAction());
+  }, [dispatch]);
+
+  const characters = useSelector((state) => state.characters);
+  let filterredCharacters = characters.slice(0, 25);
+  if (charNameInput.trim().length > 0) {
+    filterredCharacters = characters.filter((item) => (
+      item.name.toLowerCase().includes(charNameInput.toLowerCase())
+    ));
+  }
+
+  return (
+    <div className="main">
+      <div className="searchField">
+        <input type="text" value={charNameInput} placeholder="Enter your favourite Hogwarts character" onChange={inputCharNameChangeHandler} />
+      </div>
+      <div className="cards">
+        <ul>
+          {filterredCharacters.map((character) => (
+            <li className="mainContainer" key={character.id}>
+              <Link to={`/details/${character.name}`}>
+                <img src={character.image} alt={character.name} />
+                <p>{character.name}</p>
+              </Link>
+            </li>
+          ))}
+        </ul>
+      </div>
+
+    </div>
+  );
+}
```

Comment:
- [ ] You did a great job 😃 on the project, but when I see here I think the style of the page is a different little from the original design, so it will be better if you decrease the margin between the elements to make it nice 👌😊like the Behance Design as you see here the margin between them it is bigger than the Behance design, so let's try to make that like the Behance design 👌💯

> Check below to see the difference 😊⤵️
> If you can not see the image clearly just click on it to open a new tab in the browser😊

| Figma design | Your design| 
| --- | --- |
|  ![image](https://user-images.githubusercontent.com/83360636/233749815-aa47931f-997f-40d8-b97f-bc8643f6fc6a.png)  |   ![image](https://user-images.githubusercontent.com/83360636/233749784-13769825-6d3c-434d-b1b4-5233aa9f24c3.png)  |

Reply:
Hi, @osamaashraf6 if you look at my App from the mobile version you'll see that my styling is good.

Reply:
- Great job @Patfarmurs for implementing the required changes. 👍🏽 
- I' like to suggest that you fix the home page of your mobile app. Because, currently, the home page on your application doesn't align with that of Behance.

Here is what i mean 👇🏽 
|Your web app home page|
| :---:   | 
|<img width="391" alt="Screenshot 2023-04-22 at 16 11 20" src="https://user-images.githubusercontent.com/99246359/233805178-8b493e3c-185a-48db-b7ac-2dbb4de36638.png">  |